app/rtc_s1_parallel.py

```diff
@@ -0,0 +1,887 @@
+
+            if os.path.exists(geo_burst_filename):
+                rtc_s1._separate_pol_channels(geo_burst_filename,
+                                       output_burst_imagery_list,
+                                       logger, output_raster_format)
+            else:
+                for filename in output_burst_imagery_list:
+                    os.rename(filename.replace(output_dir, scratch_path), filename)
+
+                # create a vrt from the separated pol tiffs
+                geo_burst_filename +='.vrt'
+                output_imagery_list[-1] = geo_burst_filename
+                gdal.BuildVRT(geo_burst_filename, output_burst_imagery_list,
+                              options=vrt_options_mosaic)
+
+            output_file_list += output_burst_imagery_list
+
+        if save_nlooks:
+            if not flag_bursts_secondary_files_are_temporary:
+                logger.info(f'file saved: {nlooks_file}')
+            output_metadata_dict['nlooks'][1].append(nlooks_file)
+    
+        if save_rtc_anf:
+            if not flag_bursts_secondary_files_are_temporary:
+                logger.info(f'file saved: {rtc_anf_file}')
+            output_metadata_dict['rtc'][1].append(rtc_anf_file)
+
+        radar_grid_file_dict = {}
+        if flag_call_radar_grid and save_bursts:
+            rtc_s1.get_radar_grid(
+                geogrid, dem_interp_method_enum, product_prefix,
+                bursts_output_dir, imagery_extension, save_incidence_angle,
+                save_local_inc_angle, save_projection_angle,
+                save_rtc_anf_psi,
+                save_range_slope, save_dem,
+                dem_raster, radar_grid_file_dict,
+                mosaic_geogrid_dict, orbit,
+                verbose = not flag_bursts_secondary_files_are_temporary)
+            if flag_bursts_secondary_files_are_temporary:
+                # files are temporary
+                temp_files_list += list(radar_grid_file_dict.values())
+            else:
+                output_file_list += list(radar_grid_file_dict.values())
+
+        # Create burst HDF5
+        if ((save_imagery_as_hdf5 or save_metadata) and save_bursts):
+            hdf5_file_output_dir = os.path.join(output_dir, burst_id)
+            os.makedirs(hdf5_file_output_dir, exist_ok=True)
+            output_hdf5_file_burst =  os.path.join(
+                hdf5_file_output_dir, f'{product_prefix}.{hdf5_file_extension}')
+            hdf5_obj = rtc_s1.create_hdf5_file(output_hdf5_file_burst, orbit, burst, cfg)
+            rtc_s1.save_hdf5_file(
+                hdf5_obj, output_hdf5_file_burst, flag_apply_rtc,
+                clip_max, clip_min, output_radiometry_str,
+                geogrid, pol_list, geo_burst_filename, nlooks_file,
+                rtc_anf_file, layover_shadow_mask_file,
+                radar_grid_file_dict,
+                save_imagery = save_imagery_as_hdf5,
+                save_secondary_layers = save_secondary_layers_as_hdf5)
```

Comment:
After removing the snippet above, the burst HDF5 files that should be inside the parent process' output directory is not created. (e.g. `.../rtc_product_output_dir_single_burst_30m/t069_147171_iw1/rtc_product_single_burst_v3.1.h5`)
 Maybe the child processes should suppress that part, and the parent process has to run this?

Reply:
The idea is to have this parallelized, @seongsujeong . Please try to figure out what's happening and let me know if you cannot fix it.

Reply:
The flag `save_secondary_layers_as_hdf5` is the issue. we need to turn that off in child process to keep the secondary layer files as .tif files, so that we make use of them when mosaicking. Otherwise, we will get the error like below:

```
[StdErr] Traceback (most recent call last):
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/bin/rtc_s1_parallel.py", line 871, in <module>
[StdErr]     main()
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/bin/rtc_s1_parallel.py", line 866, in main
[StdErr]     run_parallel(cfg)
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/bin/rtc_s1_parallel.py", line 745, in run_parallel
[StdErr]     rtc_s1.compute_weighted_mosaic_raster(input_files, nlooks_list, output_file,
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/lib/python3.10/site-packages/rtc/mosaic_geobursts.py", line 282, in compute_weighted_mosaic_raster
[StdErr]     mosaic_dict = compute_weighted_mosaic_array(list_rtc_images, list_nlooks,
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/lib/python3.10/site-packages/rtc/mosaic_geobursts.py", line 136, in compute_weighted_mosaic_array
[StdErr]     raster_in = gdal.Open(path_rtc, gdal.GA_ReadOnly)
[StdErr]   File "/Users/jeong/miniconda3/envs/isce3_090/lib/python3.10/site-packages/osgeo/gdal.py", line 4716, in Open
[StdErr]     return _gdal.Open(*args)
[StdErr] RuntimeError: scratch_dir/temp_1676006946.225734/t069_147170_iw1//rtc_product_single_burst_v3.1_layover_shadow_mask.tif: No such file or directory
```

Reply:
I see, @seongsujeong . One way to solve that is to mosaic the secondary rasters directly from the HDF5 datasets. What do you think? Do you know how to do that? In gdal, you can use the reference: `NETCDF:<HDF5_FILE>:<DATASET_PATH>` in the place of the file name (but please check in the code if it makes sense to use that.

On a different note, `save_bursts` is `False`, you still need the children's burst values. So, you may want to force `save_bursts` in the childrens runconfig to be `True`, the flag `save_secondary_layers_as_hdf5` to be `False` and the imagery format to be `ENVI`.